src/core/thread/network_data.cpp

```diff
@@ -665,6 +676,12 @@ void NetworkData::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aM
 
     otLogInfoNetData("Server data notification acknowledged\n");
 
+    if (mLocal)
```

Comment:
Minor.

Isn't it possible that before the device gets an ACK from the leader to a notification which has been already sent it might get an advertisement? This will trigger an unwanted notification.

I think storing the timer to send will solve that issue.